boa/src/builtins/array/mod.rs

```diff
@@ -688,6 +688,9 @@ impl Array {
         let arg_c = args.len();
 
         if arg_c > 0 {
+            if len + arg_c > Number::MAX_SAFE_INTEGER as usize {
+                return context.throw_type_error("len + argCount exceeded MAX_SAFE_INTEGER");
```

Comment:
I would rephrase the message a bit, so that it's a bit more understandable. Something like "the length + the number of arguments exceed the maximum safe integer limit".